### PR TITLE
Coverity 1224619: Uninitialized pointer field

### PR DESCRIPTION
**Pull request by reveller**

### CODE REVIEW

**Comment by atsci** [comment]
RAT check *successful*! https://ci.trafficserver.apache.org/job/RAT-github/459/

**Comment by atsci** [comment]
clang format *successful*! https://ci.trafficserver.apache.org/job/clang-format-github/449/

**Comment by atsci** [comment]
Intel CC build *successful*! https://ci.trafficserver.apache.org/job/icc-github/572/

**Comment by atsci** [comment]
Linux build *successful*! https://ci.trafficserver.apache.org/job/linux-github/2038/

**Comment by atsci** [comment]
AU check *successful*! https://ci.trafficserver.apache.org/job/autest-github/446/

**Comment by atsci** [comment]
FreeBSD11 build *successful*! https://ci.trafficserver.apache.org/job/freebsd-github/2142/

**Comment by atsci** [comment]
clang-analyzer build *successful*! https://ci.trafficserver.apache.org/job/clang-analyzer-github/703/

File: proxy/http2/Http2ClientSession.h [hunk 1]
```diff
@@ -307,26 +307,26 @@ class Http2ClientSession : public ProxyClientSession
...
-  int64_t total_write_len;
-  SessionHandler session_handler;
-  NetVConnection *client_vc;
-  MIOBuffer *read_buffer;
-  IOBufferReader *sm_reader;
-  MIOBuffer *write_buffer;
-  IOBufferReader *sm_writer;
-  Http2FrameHeader current_hdr;
+  int64_t total_write_len        = 0;
+  SessionHandler session_handler = 0;
```
Comment (bryancall) [inline proxy/http2/Http2ClientSession.h]: should this be = nullptr?

Comment (zwoop) [inline proxy/http2/Http2ClientSession.h]: Yeh, I think it should be nullptr.

**Comment by atsci** [comment]
clang format *successful*! https://ci.trafficserver.apache.org/job/clang-format-github/464/

**Comment by atsci** [comment]
RAT check *successful*! https://ci.trafficserver.apache.org/job/RAT-github/473/

**Comment by atsci** [comment]
FreeBSD11 build *successful*! https://ci.trafficserver.apache.org/job/freebsd-github/2156/

**Comment by atsci** [comment]
Linux build *successful*! https://ci.trafficserver.apache.org/job/linux-github/2052/

**Comment by atsci** [comment]
AU check *failed*! https://ci.trafficserver.apache.org/job/autest-github/460/

**Comment by atsci** [comment]
Intel CC build *successful*! https://ci.trafficserver.apache.org/job/icc-github/587/

**Comment by atsci** [comment]
clang-analyzer build *successful*! https://ci.trafficserver.apache.org/job/clang-analyzer-github/716/

**Comment by PSUdaemon** [comment]
AU test failed.
[approve ci]

**Comment by atsci** [comment]
clang format *successful*! https://ci.trafficserver.apache.org/job/clang-format-github/480/

**Comment by atsci** [comment]
RAT check *successful*! https://ci.trafficserver.apache.org/job/RAT-github/489/

**Comment by atsci** [comment]
FreeBSD11 build *successful*! https://ci.trafficserver.apache.org/job/freebsd-github/2172/

**Comment by atsci** [comment]
Intel CC build *successful*! https://ci.trafficserver.apache.org/job/icc-github/603/

**Comment by atsci** [comment]
AU check *successful*! https://ci.trafficserver.apache.org/job/autest-github/478/

**Comment by atsci** [comment]
Linux build *successful*! https://ci.trafficserver.apache.org/job/linux-github/2069/

**Comment by atsci** [comment]
clang-analyzer build *successful*! https://ci.trafficserver.apache.org/job/clang-analyzer-github/732/

**Pull request merged**